tools/releaseBuild/azureDevOps/templates/release-PublishToWinget.yml

```diff
@@ -0,0 +1,28 @@
+parameters:
+  - name: skipPublish
+    default: false
+    type: boolean
+
+steps:
+  condition: and(eq('${{ parameters.skipPublish }}', 'false'), succeeded())
+  - name: Checkout PowerShell
+    uses: actions/checkout@v3
+    with:
+      repository: PowerShell/PowerShell
+      path: './powershell'
+  - name: Release latest PowerShell package msi installers to winget-pkgs
+    id: release_info
+    pwsh: |
+      # Get latest release from Github and release metadata
+      $targetRelease = Invoke-RestMethod -Uri https://api.github.com/repos/PowerShell/PowerShell/releases/latest
+      $installerUrls = $targetRelease | Select-Object -ExpandProperty assets | Where-Object -Property name -like "PowerShell*msi" | Select-Object -ExpandProperty browser_download_url
+      $releaseTag = $targetRelease.tag_name
+      echo $releaseTag
+      $version = $releaseTag.Trim("v")
+      echo $version
+      $pkgId = $version.Contains("-") ? "Microsoft.PowerShell.Preview" : "Microsoft.PowerShell"
+      echo $pkgId
+
+      # Update package using wingetcreate
+      Invoke-WebRequest https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
+      .\wingetcreate.exe update $pkgId --version $version --urls $installerUrls --submit --token ${{ secrets.PR_PAT }}
```

Comment:
This won't work if the previous manifest contains the MSIX installers. Winget-Create works when the number of new installer URLs match the number in the previous (base) manifest (the msixbundle link needs to be provided only once tho). My understanding is that the msixbundle is published sometime (1-2 days??) after the MSI releases; If this action triggers when the MSIs are published to GitHub and the previous winget manifest contains MSIX installer nodes as well, the current implementation of winget-create will fail to generate the PR.

The solution I can think of is one of these:

- Address this in winget-create's implementation to allow automated PRs with less number of installer nodes/URLs
- Release MSIXBundle at the same time as MSIs and update the action to generate the complete manifest with msixbundle included. Or wait till msixbundle URL is available on GitHub and run the action as soon as that happens (but that wouldn't be reasonable since the point of this automation is to get this package out ASAP)
- (I know you mentioned this but I'm unaware of its progress) Release MSIs for ARM architecture that are published at the same time as x64/x86. That way we wouldn't need to add msix in winget's manifest and the automation will work correctly
- Don't add MSIX for the "latest" winget manifest, add them for the previous version after the automated PR (that also doesn't sound reasonable to me. I'm just writing anything that can work 😄 )
- Try another manifest authoring tool? There are a couple community developed tools that can handle the task of automated PRs that may not be as strict as winget-create